File: 프로그래머스/2/138476. 귤 고르기/귤 고르기.py
def solution(k, tangerine):
    size_count_map = {}
    
    for size in tangerine:
        if size in size_count_map:
            size_count_map[size] += 1
        else:
            size_count_map[size] = 1
    
    count_list = sorted(size_count_map.values(), reverse=True)
    
    total = 0
    kinds = 0
    
    for count in count_list:
        total += count
        kinds += 1
        if total >= k:
            return kinds

    return kinds

File: 프로그래머스/2/138476. 귤 고르기/귤 고르기.java
import java.util.*;
class Solution {
    public int solution(int k, int[] tangerine) {
        int answer = 0;
        
        int[] count = new int[10000001];
        int maxSize = 0;
        
        for (int size : tangerine) {
            count[size]++;
            maxSize = Math.max(maxSize, size);
        }
        
        List<Integer> countList = new ArrayList<>();
        
        for (int i = 1; i <= maxSize; i++) {
            if (count[i] > 0) {
                countList.add(count[i]);
            }
        }
        
        countList.sort(Collections.reverseOrder());
        
        int sum = 0;
        for (int cnt : countList) {
            sum += cnt;
            answer++;
            if (sum >= k) {
                break;
            }
        }

        return answer;
    }
}
